routing/pathfind.go

```diff
@@ -189,6 +191,23 @@ func newRoute(sourceVertex route.Vertex,
 					"custom records")
 			}
 			customRecords = finalHop.records
+
+			// If we're attaching a payment addr but the receiver
+			// doesn't support both TLV and payment addrs, fail.
+			payAddr := supports(lnwire.PaymentAddrOptional)
+			if (!tlvPayload || !payAddr) &&
```

Comment:
Isn't `tlvPayload` implicitly supported after we've validated the feature bits?

Reply:
good point, fixed